gptcache/manager/factory.py

```diff
@@ -1,8 +1,91 @@
+import os
 from typing import Union, Callable
 from gptcache.manager.data_manager import SSDataManager, MapDataManager
 from gptcache.manager import CacheBase, VectorBase, ObjectBase
 
 
+def manager_factory(manager="Map",
+                    data_dir="./",
+                    max_size = 1000,
+                    clean_size = None,
+                    eviction: str = "LRU",
+                    get_data_container: Callable = None,
+                    scalar_params=None,
+                    vector_params=None,
+                    object_params=None):
+
+    """Factory of `DataManager`.
+       By using this factory method, you only need to specify the root directory of the data,
+       and it can automatically manage all the local files.
+
+    :param manager: Type of DataManager. Supports: Map, or {scalar_name},{vector_name}
+                    or {scalar_name},{vector_name},{object_name}
+    :type manager: str
+    :param data_dir: Root path for data storage.
+    :type data_dir: str
+    :param max_size: the max size for the cache, defaults to 1000.
+    :type max_size: int
+    :param clean_size: the size to clean up, defaults to `max_size * 0.2`.
+    :type clean_size: int
+    :param eviction: the eviction policy, it is support "LRU" and "FIFO" now, and defaults to "LRU".
+    :type eviction:  str
+    :param get_data_container: a Callable to get the data container, defaults to None.
+    :type get_data_container:  Callable
+
+    :param scalar_params: Params of scalar storage.
+    :type scalar_params:  dict
+
+    :param vector_params: Params of vector storage.
+    :type vector_params:  dict
+
+    :param object_params: Params of object storage.
+    :type object_params:  dict
+
+    :return: SSDataManager or MapDataManager.
+
+    Example:
+        .. code-block:: python
+
+            from gptcache.manager import manager_factory
+
+            data_manager = manager_factory("sqlite,faiss", data_root="./workspace", vector_params={"dimension": 128})
+    """
+
+    if manager == "Map":
+        return MapDataManager(os.path.join(data_dir, "data_map.txt"), max_size, get_data_container)
+
+    db_infos = manager.split(",")
+    if len(db_infos) not in [2, 3]:
+        raise RuntimeError("Error manager format: %s, the correct is \"{scalar},{vector},{object}\", object is optional" % manager)
+
+    if len(db_infos) == 2:
+        db_infos.append(None)
+    scalar, vector, obj = db_infos
+
+    if scalar_params is None:
+        scalar_params = {}
+    if scalar == "sqlite":
+        scalar_params["sql_url"] = "sqlite:///" + os.path.join(data_dir, "sqlite.db")
+    s = CacheBase(name=scalar, **scalar_params)
+
+    if vector_params is None:
+        vector_params = {}
+    if vector == "faiss":
+        vector_params["index_path"] = os.path.join(data_dir, "faiss.bin")
```

Comment:
the default is `faiss.index`, if you want to change it, please change other relative codes

Reply:
fixed